app/controllers/api/account.php

```diff
@@ -3703,6 +3706,99 @@
+                ->setParam('{{project}}', $project->getAttribute('name'))
+                ->setParam('{{user}}', $user->getAttribute('name'));
+
+            $message->setParam('{{body}}', $body->render(), escapeHtml: false);
+        } else {
+            $message = Template::fromFile(__DIR__ . '/../../config/locale/templates/email-inner-base.tpl');
+
+            $message->setParam('{{body}}', $locale->getText("emails.mfaCreated.body"), escapeHtml: false);
+        }
+
+        $message
+            ->setParam('{{hello}}', $locale->getText("emails.mfaCreated.hello"))
+            ->setParam('{{footer}}', $locale->getText("emails.mfaCreated.footer"))
+            ->setParam('{{thanks}}', $locale->getText("emails.mfaCreated.thanks"))
+            ->setParam('{{signature}}', $locale->getText("emails.mfaCreated.signature"));
+
+        $body = $message->render();
+
+        $smtp = $project->getAttribute('smtp', []);
+        $smtpEnabled = $smtp['enabled'] ?? false;
+
+        $senderEmail = System::getEnv('_APP_SYSTEM_EMAIL_ADDRESS', APP_EMAIL_TEAM);
+        $senderName = System::getEnv('_APP_SYSTEM_EMAIL_NAME', APP_NAME . ' Server');
+        $replyTo = "";
+
+        if ($smtpEnabled) {
+            if (!empty($smtp['senderEmail'])) {
+                $senderEmail = $smtp['senderEmail'];
+            }
+            if (!empty($smtp['senderName'])) {
+                $senderName = $smtp['senderName'];
+            }
+            if (!empty($smtp['replyTo'])) {
+                $replyTo = $smtp['replyTo'];
+            }
+
+            $queueForMails
+                ->setSmtpHost($smtp['host'] ?? '')
+                ->setSmtpPort($smtp['port'] ?? '')
+                ->setSmtpUsername($smtp['username'] ?? '')
+                ->setSmtpPassword($smtp['password'] ?? '')
+                ->setSmtpSecure($smtp['secure'] ?? '');
+
+            if (!empty($customTemplate)) {
+                if (!empty($customTemplate['senderEmail'])) {
+                    $senderEmail = $customTemplate['senderEmail'];
+                }
+                if (!empty($customTemplate['senderName'])) {
+                    $senderName = $customTemplate['senderName'];
+                }
+                if (!empty($customTemplate['replyTo'])) {
+                    $replyTo = $customTemplate['replyTo'];
+                }
+
+                $body = $customTemplate['message'] ?? '';
+                $subject = $customTemplate['subject'] ?? $subject;
+            }
+
+            $queueForMails
+                ->setSmtpReplyTo($replyTo)
+                ->setSmtpSenderEmail($senderEmail)
+                ->setSmtpSenderName($senderName);
+        }
+
+        $emailVariables = [
+            'owner' => $user->getAttribute('name'),
+            'direction' => $locale->getText('settings.direction'),
+            'user' => $user->getAttribute('name'),
+            'project' => $project->getAttribute('name')
+        ];
```

Comment:
redirect is required in email-inner-base.tpl 😬

Reply:
We shouldn't be using redirect should we? We don't know where the user wants to redirect. It's up to them to specify it in their template otherwise it'll look like so:

![image](https://github.com/user-attachments/assets/e30ec24a-3ae0-4f8b-8fc0-ab7e06410ad8)

Reply:
> We shouldn't be using redirect should we? We don't know where the user wants to redirect.

Correct.

> It's up to them to specify it in their template

But developers on the free plan can't set a custom template.

Reply:
Split them :+1: